generator/ProtocolTestsGenerator/README.md

```diff
@@ -0,0 +1,39 @@
+# Protocol Test Generator
+
+### Project Structure
+The `ProtocolTestsGenerator` folder consists of 3 gradle projects. The `smithy-dotnet-codegen` project contains
+the core logic for generating the protocol tests. At present, only protocol test generation via smithy test files is supported at this time.
+Please note that this is not a full smithy code generator.
+
+The `smithy-dotnet-codegen-test` project consists of several test smithy models that can be used to test the core
+`smithy-dotnet-codegen` project. 
+
+The `smithy-dotnet-protocol-test` project is where the protocol tests will be output under `build/smithyprojections/smithy-dotnet-protocol-test`
+
+## Setup
+This project makes use of `gradlew`, so no JDK dependency is needed to build the project using the `./gradlew` command.
+Simply clone the repository and `cd` into the `ProtocolTestsGenerator` folder. The gradle wrapper will invoke a declared version
+of gradle or download it beforehand if necessary.
+
+## Running the Protocol Test Generator
+Once in the `ProtocolTestsGenerator` folder.
+1. Navigate to `smithy-dotnet-protocol-test` directory
+```
+cd smithy-dotnet-protocol-test
+```
+2. Build the project
+```
+./gradlew :smithy-dotnet-protocol-test:build
+```
+
+## Running the Test Project
+1. Navigate to `smithy-dotnet-codegen-test`
+```
+cd smithy-dotnet-codegen-test
+```
+2. Build the project
+```
+./gradlew :smithy-dotnet-codegen-test:build
```

Comment:
I'm getting the following error on this step. Keep in mind per my previous comment I'm using `../gradlew`.

```
Projection source failed: software.amazon.smithy.model.node.ExpectationNotMetException: Missing expected member `package`. (C:\codebase\v3\AWSDotNetPublic\generator\ProtocolTestsGenerator\smithy-dotnet-codegen-test\smithy-build.json [4, 27])
software.amazon.smithy.model.node.ExpectationNotMetException: Missing expected member `package`. (C:\codebase\v3\AWSDotNetPublic\generator\ProtocolTestsGenerator\smithy-dotnet-codegen-test\smithy-build.json [4, 27])
```

Reply:
This error is because I updated the `DotnetSettings.java` file to expect different values in the `smithy-build.json` and I forgot to update the test files. I updated the smithy-build.json for the test project. However, running the test project will throw an exception saying "The given protocol is not supported" which is because the core smithy code generator is configured to specifically generate protocol tests. I added a note in the README which warns of this. Since we're not a full smithy code generator, the test project needs some tweaking to work normally.